src/CabanaPD_Input.hpp

```diff
@@ -95,6 +95,9 @@ class Inputs
         if ( inputs.contains( "system_size" ) )
         {
             auto size = inputs["system_size"]["value"];
+            if ( size != 3 )
```

Comment:
As opposed to Line 116, size here is not the size of an array but the size of the system. So, I believe "if ( size != 3 )" is wrong.